### PR TITLE
Use api for adding metadata and publishing modules

### PR DESCRIPTION
**Pull request by noopurmath3**
- Append random string for product name
- User api endpoint for adding metadata and publish

### CODE REVIEW

File: fixtures/fixture.py [hunk 1]
```diff
@@ -127,6 +132,16 @@ def setup_test_products():
                response.status_code, any_of(equal_to(201), equal_to(200)))
 
 
+def get_product_id():
+    # Get ID of the product created
+    path_to_product_node = url + "content/products/" + constants.product_name_uri
+    path_to_version = path_to_product_node + "/versions/{}".format(constants.product_version)
+    path = path_to_version+".json"
```
Comment (anarang) [inline fixtures/fixture.py:139]: minor change: please add a space before and after "+"

File: helpers/constants.py [hunk 2]
```diff
@@ -52,6 +52,9 @@
 module_not_found = "Module not found"
 version_url_fragment_warning = "Allowed input for ulrFragment: alphanumeric, hyphen, period and underscore"
 new_product_url_fragment = "Test UrlFragment"
+random = utilities.generate_random_string(4)
```
Comment (anarang) [inline helpers/constants.py:55]: @noopurmath3 this is not the right place to use a function, this file contains only the constants, we cannot have a random string generator in the middle of constants. please use this function somewhere in other modules or maybe a separate function to do the same, refer api-tests suite for consistency.

**Review by anarang — APPROVED** [review]
@noopurmath3 looks good! thanks. :)

**Review by pbadak — APPROVED** [review]
@noopurmath3 Looks good to me .Thanks :)

File: helpers/constants.py (hunk 2 above)
Comment (noopurmath3) [reply helpers/constants.py:55]: Marking this resolved as per conversations on chat

**Pull request merged**